react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      id: 1
+    };
+    this.handleClickLeft = this.handleClickLeft.bind(this);
+    this.handleClickRight = this.handleClickRight.bind(this);
+    this.stepper = this.stepper.bind(this);
+    this.componentDidMount = this.componentDidMount.bind(this);
+  }
+
+  handleClickLeft() {
+    clearInterval(this.timerID);
+    if (this.state.id === 1) {
+      this.setState({ id: 1 });
+    } else {
+      this.setState({ id: this.state.id - 1 });
+    }
+    this.componentDidMount();
```

Comment:
Clever! But you should never call the lifecycle methods directly, those are for React to manage internally. a better solution would be to define a `startInterval` method (or whatever you want to call it). Then call it here:

```jsx
 startInterval() [
   this.timerID = setInterval(
      () => this.stepper(),
      3000
    );
  }
```